agent/exec/container/controller.go

```diff
@@ -194,6 +195,12 @@ func (r *controller) Wait(pctx context.Context) error {
 				// If we get here, something has gone wrong but we want to exit
 				// and report anyways.
 				return ErrContainerDestroyed
+			case "health_status: unhealthy":
```

Comment:
There is a healthcheck `Interval` in the arguments you can provide, the point of which is that the healthcheck ought to run periodically.

My concern is that this `"health_status: unhealthy"` event you're checking for is the one sent when the container is actually dead. The case we'd like to focus on with healthchecks is when the container is alive but in an unfavorable state (like deadlocked). Can you confirm that the event you're getting is the one we want?